easybuild/framework/easyconfig/easyconfig.py

```diff
@@ -539,13 +540,28 @@ def parse(self):
 
         run_hook(PARSE, hooks, args=[self], msg=parse_hook_msg)
 
+        # create a list of all options that are actually going to be iterated over
+        # builddependencies are always a list, need to look deeper down below
+        self.iterate_options = [opt for opt in ITERATE_OPTIONS
+                                if opt != 'builddependencies' and isinstance(self[opt], (list, tuple))]
```

Comment:
@bartoldeman I only realize now that this is done too early...

We're assuming here that once the easyconfig file is parsed, we know whether which easyconfig parameters are to be iterated over. That breaks backwards compatibility in some sense, since before this change we had the ability to tweak the easyconfig parameter values before they were actually being used...

And that's exactly what we're doing with `FFTW`, where we define the list of configure options to iterate over in `self.cfg['configopts']` via `run_all_steps`...
That value is not picked up for defining `self.iterate_options`, and hence stuff is broken.

I think we'll need to postpone this to avoid breaking backwards compatibility (yet still handle `builddependencies` here)...

Reply:
problem should be fixed with #2810